### PR TITLE
Extra stop phrases

### PR DESCRIPTION
**Pull request by Alcray**

### CODE REVIEW

File: nemo_skills/inference/generate.py [hunk 1]
```diff
@@ -213,10 +218,11 @@ def generate(cfg: GenerateSolutionsConfig):
                 if cfg.multi_turn_key is None:
                     outputs = llm.generate(
                         prompts=[prompt.fill(dp) for dp in data_points],
-                        stop_phrases=prompt.stop_phrases,
+                        stop_phrases = (prompt.stop_phrases if prompt.stop_phrases is None else (prompt.stop_phrases + extra_stop_phrases)),
```
Comment (Kipok) [inline nemo_skills/inference/generate.py:221]: why such condition? If it's None we should still take the extra stop phrases, right?

**Comment by Kipok** [comment]
Can you please give an example how this is passing through command line? Is it too cumbersome to specify as a list or not too bad?

**Comment by Alcray** [comment]
exaple of call
```
python nemo_skills/pipeline/generate.py \
    --partition interactive \
    --cluster=draco-oci \
    --server_type=trtllm \
    --model=/trt_models/llama3.1-405b-instruct \
    --server_gpus=8 \
    --server_nodes 2 \
    --output_dir=/openmath2data-ref/nat-to-lean4/trt_llm_test/llama3.1-405b-instruct \
    --eval_args="++eval_type=lean4-statement-with-header" \
    ++dataset=math \
    ++split=test \
    ++prompt_config=lean4/nat-to-lean4-math-llama \
    ++examples_type=math_to_lean4_fewshot \
    ++prompt_template=llama3-instruct-nosys \
    ++extra_stop_phrases=["sorry"] \
    ++skip_filled=False \
    ++inference.tokens_to_generate=512 \
    ++max_samples=10
    ```
    
    For me it sims very consistent to the both command and config style. I thing it might haver some confusions if we are not passing it as a list

File: nemo_skills/inference/generate.py (hunk 1 above)
Comment (Alcray) [reply nemo_skills/inference/generate.py:221]: It sims like mt-bench test in test_generations expect None as an input and are failing, when trying to pass empty array

File: nemo_skills/inference/generate.py [hunk 2]
```diff
@@ -213,10 +226,11 @@ def generate(cfg: GenerateSolutionsConfig):
                 if cfg.multi_turn_key is None:
                     outputs = llm.generate(
                         prompts=[prompt.fill(dp) for dp in data_points],
-                        stop_phrases=prompt.stop_phrases,
+                        stop_phrases = combine_stop_phrases(prompt.stop_phrases, extra_stop_phrases),
```
Comment (Kipok) [inline nemo_skills/inference/generate.py:229]:
```suggestion
                        stop_phrases=combine_stop_phrases(prompt.stop_phrases, extra_stop_phrases),
```

File: nemo_skills/inference/generate.py [hunk 3]
```diff
@@ -242,7 +256,7 @@ def generate(cfg: GenerateSolutionsConfig):
                                 prompt.fill(turn_data_points[dp_index], multi_turn_key=cfg.multi_turn_key)
                                 for dp_index in dp_indices
                             ],
-                            stop_phrases=prompt.stop_phrases,
+                            stop_phrases = combine_stop_phrases(prompt.stop_phrases, extra_stop_phrases),
```
Comment (Kipok) [inline nemo_skills/inference/generate.py:259]:
```suggestion
                            stop_phrases=combine_stop_phrases(prompt.stop_phrases, extra_stop_phrases),
```

**Review by Kipok — APPROVED** [review]

**Pull request merged**